spirv_glsl.cpp

```diff
@@ -3306,40 +3336,47 @@ std::string CompilerGLSL::flattened_access_chain(uint32_t base, const uint32_t *
                                                  const SPIRType &target_type, uint32_t offset)
 {
 	if (!target_type.array.empty())
-	{
 		SPIRV_CROSS_THROW("Access chains that result in an array can not be flattened");
-	}
 	else if (target_type.basetype == SPIRType::Struct)
-	{
 		return flattened_access_chain_struct(base, indices, count, target_type, offset);
-	}
 	else if (target_type.columns > 1)
-	{
 		return flattened_access_chain_matrix(base, indices, count, target_type, offset);
-	}
 	else
-	{
 		return flattened_access_chain_vector_scalar(base, indices, count, target_type, offset);
-	}
 }
 
 std::string CompilerGLSL::flattened_access_chain_struct(uint32_t base, const uint32_t *indices, uint32_t count,
                                                         const SPIRType &target_type, uint32_t offset)
 {
 	std::string expr;
 
-	expr += type_to_glsl(target_type);
+	expr += type_to_glsl_constructor(target_type);
 	expr += "(";
 
+	// Need to have an extra indices for the member so we can find matrix strides.
```

Comment:
The comment should say "whether matrix members need transposing" - AFAICS matrix strides are not retrieved here